src/saml2/mdstore.py

```diff
@@ -829,21 +832,35 @@ def load(self, *args, **kwargs):
         # Do nothing
         pass
 
+    def fetch_metadata(self, item):
+        mdx_url = "%s/entities/%s" % (self.url, self.entity_transform(item))
+        response = requests.get(mdx_url, headers={
+            'Accept': SAML_METADATA_CONTENT_TYPE})
+        if response.status_code == 200:
+            _txt = response.content
+            if self.parse_and_check_signature(_txt):
+                if self.freshness_period:
+                    curr_time = str_to_time(instant())
+                    self.expiration_date[item] = add_duration(
+                        curr_time, self.freshness_period)
+                return self.entity[item]
+        else:
+            logger.info("Response status: %s", response.status_code)
+        raise KeyError
+
     def __getitem__(self, item):
-        try:
-            return self.entity[item]
-        except KeyError:
-            mdx_url = "%s/entities/%s" % (self.url, self.entity_transform(item))
-            response = requests.get(mdx_url, headers={
-                'Accept': SAML_METADATA_CONTENT_TYPE})
-            if response.status_code == 200:
-                _txt = response.content
-
-                if self.parse_and_check_signature(_txt):
-                    return self.entity[item]
+        if item in self.entity:
+            if self.freshness_period and not \
```

Comment:
Minor but I would make this a function named "is_fresh" or something to make the condition a bit clearer.